src/components/toggle-button-group/ToggleButtonGroup.tsx

```diff
@@ -102,6 +102,8 @@ function ToggleButtonItem(props: ToggleButtonItemProps) {
 }
 
 const StyledRoot = styled(RadixToggleGroup.Root)`
+  isolation: isolate; // Prevents the z-index changes of the items inside from affecting layers outside the ToggleButtonGroup
```

Comment:
The comment is a bit vague, what specific z-index conflict is this solving?

Reply:
Updated the comment to be more specific.

But I think it's juste a general rule of thumb, if we use z-index in one of our component to change the layer order of that component rendering, these layer order changes must be scoped to that component only, and we can achieve that by creating a new stacking context at the root of the component.

Non of our fixed z-index in the library should have effect in the outside world.